designer/server/src/main/scala/pl/touk/nussknacker/ui/api/NuDesignerErrorToHttp.scala

```diff
@@ -16,26 +16,30 @@ object NuDesignerErrorToHttp extends LazyLogging with FailFastCirceSupport {
   def nuDesignerErrorHandler: ExceptionHandler = {
     import akka.http.scaladsl.server.Directives._
     ExceptionHandler { case NonFatal(e) =>
-      complete(errorToHttpResponse(e))
+      complete(errorToHttpResponse(unwrapException(e)))
     }
   }
 
-  private def errorToHttpResponse: PartialFunction[Throwable, HttpResponse] = {
+  def unwrapException(e: Throwable): Throwable =
+    if (Option(e.getCause).nonEmpty) {
```

Comment:
IMO it's not a good approach. Let's say that someone throw NuDesignerError with some cause. Will be it handled correctly? I thought about:
```
e match {
case SomeSpecificClassNameThatWeAreSureThatWeWantExtractCause => e.getCasue
case _ => e
}
```
By `SomeSpecificClassNameThatWeAreSureThatWeWantExtractCause` you said something about BoxedException? Am I right?